x/migration/keeper/msg_server_claim_morse_supplier.go

```diff
@@ -269,28 +317,58 @@ func (k msgServer) ClaimMorseSupplier(
 		return nil, err
 	}
 
-	claimedSupplierStake := morseClaimableAccount.GetSupplierStake()
-	claimedUnstakedBalance := morseClaimableAccount.GetUnstakedBalance()
+	events := []cosmostypes.Msg{
+		// Emit an event which signals that the morse account has been claimed.
+		&migrationtypes.EventMorseSupplierClaimed{
+			MorseNodeAddress:     msg.GetMorseNodeAddress(),
+			MorseOutputAddress:   morseClaimableAccount.GetMorseOutputAddress(),
+			ClaimSignerType:      claimSignerType,
+			ClaimedBalance:       claimedUnstakedBalance,
+			ClaimedSupplierStake: claimedSupplierStake,
+			SessionEndHeight:     sessionEndHeight,
+			Supplier:             supplier,
+		},
+	}
 
-	// Emit an event which signals that the morse account has been claimed.
-	event := migrationtypes.EventMorseSupplierClaimed{
-		MorseNodeAddress:     msg.GetMorseNodeAddress(),
-		MorseOutputAddress:   morseClaimableAccount.GetMorseOutputAddress(),
-		ClaimSignerType:      claimSignerType,
-		ClaimedBalance:       claimedUnstakedBalance,
-		ClaimedSupplierStake: claimedSupplierStake,
-		SessionEndHeight:     sessionEndHeight,
-		Supplier:             supplier,
+	// If the claimed supplier is still unbonding, emit an unbonding begin event.
+	if !morseClaimableAccount.UnstakingCompletionTime.IsZero() {
```

Comment:
Would love to see some tests.

Per the comments in the file where this function is defined and how it's used, I find it hard to grok.

Reply:
Consider adding ANOTHER helper that wraps `UnstakingCompletionTime` called something like `ShouldStartAutoUnstakingMorseSupplier` for clarity.